### PR TITLE
add sklearn runner & HistGradientBoost model

### LINKED ISSUE #44: HistGradientBooster: train-stage
- [ ] Output von train_stage:
     - test-predictions as probabilites and ground_truth in pandas-dataframe [y_true and y_pred]
     - export model
     - Optional: loss, accuracy and f1 macro over epochs

- [ ] Stage: Segmentation -> FFT -> PCA -> HistGradientBooster

### PR DESCRIPTION
**Pull request by yvokeller**

### CODE REVIEW

**Review by florinbarbisch — APPROVED** [review]

**Comment by ghost** [comment]
# RUN: 2023-04-26 11-12-29

## Hyperparameters

| Path        | Param                                              | HEAD^   | workspace                      |
|-------------|----------------------------------------------------|---------|--------------------------------|
| params.yaml | hist_gradient_boost.data.type                      | -       | Sklearn                        |
| params.yaml | hist_gradient_boost.model                          | -       | HistGradientBoostingClassifier |
| params.yaml | hist_gradient_boost.model_hparams.learning_rate    | -       | 0.1                            |
| params.yaml | hist_gradient_boost.model_hparams.max_bins         | -       | 255                            |
| params.yaml | hist_gradient_boost.model_hparams.max_iter         | -       | 2000                           |
| params.yaml | hist_gradient_boost.model_hparams.max_leaf_nodes   | -       | 31                             |
| params.yaml | hist_gradient_boost.model_hparams.min_samples_leaf | -       | 20                             |

## Metrics

| Path                             | Metric   | HEAD^   | workspace   | Change   |
|----------------------------------|----------|---------|-------------|----------|
| hist_gradient_boost/metrics.json | accuracy | -       | 0.37634     | -        |
| hist_gradient_boost/metrics.json | step     | -       | 0           | -        |

# Plots

TODO

**Comment by ghost** [comment]
# RUN: 2023-04-26 11-18-59

## Hyperparameters

| Path        | Param                                              | HEAD^                          | workspace                      |
|-------------|----------------------------------------------------|--------------------------------|--------------------------------|
| params.yaml | dvclive.data.batch_size                            | 64                             | -                              |
| params.yaml | dvclive.data.input_size                            | 72                             | -                              |
| params.yaml | dvclive.data.num_workers                           | 0                              | -                              |
| params.yaml | dvclive.data.output_size                           | 6                              | -                              |
| params.yaml | dvclive.data.train_val_split                       | 0.8                            | -                              |
| params.yaml | dvclive.data.type                                  | Tabular                        | Sklearn                        |
| params.yaml | dvclive.model                                      | MLP                            | HistGradientBoostingClassifier |
| params.yaml | dvclive.model_hparams.max_bins                     | -                              | 255                            |
| params.yaml | dvclive.model_hparams.max_iter                     | -                              | 2000                           |
| params.yaml | dvclive.model_hparams.max_leaf_nodes               | -                              | 31                             |
| params.yaml | dvclive.model_hparams.min_samples_leaf             | -                              | 20                             |
| params.yaml | dvclive.model_hparams.momentum                     | 0.9                            | -                              |
| params.yaml | dvclive.model_hparams.num_epochs                   | 10                             | -                              |
| params.yaml | dvclive.model_hparams.weight_decay                 | 0.0005                         | -                              |
| params.yaml | hist_gradient_boost.data.type                      | Sklearn                        | -                              |
| params.yaml | hist_gradient_boost.model                          | HistGradientBoostingClassifier | -                              |
| params.yaml | hist_gradient_boost.model_hparams.learning_rate    | 0.1                            | -                              |
| params.yaml | hist_gradient_boost.model_hparams.max_bins         | 255                            | -                              |
| params.yaml | hist_gradient_boost.model_hparams.max_iter         | 2000                           | -                              |
| params.yaml | hist_gradient_boost.model_hparams.max_leaf_nodes   | 31                             | -                              |
| params.yaml | hist_gradient_boost.model_hparams.min_samples_leaf | 20                             | -                              |

## Metrics

| Path                             | Metric           | HEAD^   | workspace   | Change   |
|----------------------------------|------------------|---------|-------------|----------|
| dvclive/metrics.json             | acc              | -       | 0.37634     | -        |
| dvclive/metrics.json             | epoch            | 10      | -           | -        |
| dvclive/metrics.json             | f1               | -       | 0.29114     | -        |
| dvclive/metrics.json             | step             | 890     | 0           | -890     |
| dvclive/metrics.json             | test.epoch.acc   | 0.38351 | -           | -        |
| dvclive/metrics.json             | test.epoch.f1    | 0.31615 | -           | -        |
| dvclive/metrics.json             | test.step.acc    | 0.38351 | -           | -        |
| dvclive/metrics.json             | test.step.f1     | 0.31615 | -           | -        |
| dvclive/metrics.json             | train.epoch.acc  | 0.96381 | -           | -        |
| dvclive/metrics.json             | train.epoch.f1   | 0.9572  | -           | -        |
| dvclive/metrics.json             | train.epoch.loss | 0.10953 | -           | -        |
| dvclive/metrics.json             | train.step.acc   | 1.0     | -           | -        |
| dvclive/metrics.json             | train.step.f1    | 1.0     | -           | -        |
| dvclive/metrics.json             | train.step.loss  | 0.01936 | -           | -        |
| dvclive/metrics.json             | val.epoch.acc    | 0.91378 | -           | -        |
| dvclive/metrics.json             | val.epoch.f1     | 0.906   | -           | -        |
| dvclive/metrics.json             | val.epoch.loss   | 0.33504 | -           | -        |
| dvclive/metrics.json             | val.step.acc     | 0.85714 | -           | -        |
| dvclive/metrics.json             | val.step.f1      | 0.66667 | -           | -        |
| dvclive/metrics.json             | val.step.loss    | 0.39189 | -           | -        |
| hist_gradient_boost/metrics.json | acc              | 0.37634 | -           | -        |
| hist_gradient_boost/metrics.json | f1               | 0.29114 | -           | -        |
| hist_gradient_boost/metrics.json | step             | 0       | -           | -        |

# Plots

TODO

**Review by florinbarbisch — CHANGES_REQUESTED** [review]
image
ich bi dr meinig, dass d metrike vergliichbar müese sii (uf dr gliiche ziile)

**Review by florinbarbisch — APPROVED** [review]

**Comment by ghost** [comment]
# RUN: 2023-04-26 14-19-40

## Hyperparameters

| Path        | Param                                  | HEAD^   | workspace                      |
|-------------|----------------------------------------|---------|--------------------------------|
| params.yaml | dvclive.data.batch_size                | 64      | -                              |
| params.yaml | dvclive.data.input_size                | 72      | -                              |
| params.yaml | dvclive.data.num_workers               | 0       | -                              |
| params.yaml | dvclive.data.output_size               | 6       | -                              |
| params.yaml | dvclive.data.train_val_split           | 0.8     | -                              |
| params.yaml | dvclive.data.type                      | Tabular | Sklearn                        |
| params.yaml | dvclive.model                          | MLP     | HistGradientBoostingClassifier |
| params.yaml | dvclive.model_hparams.max_bins         | -       | 255                            |
| params.yaml | dvclive.model_hparams.max_iter         | -       | 2000                           |
| params.yaml | dvclive.model_hparams.max_leaf_nodes   | -       | 31                             |
| params.yaml | dvclive.model_hparams.min_samples_leaf | -       | 20                             |
| params.yaml | dvclive.model_hparams.momentum         | 0.9     | -                              |
| params.yaml | dvclive.model_hparams.num_epochs       | 10      | -                              |
| params.yaml | dvclive.model_hparams.weight_decay     | 0.0005  | -                              |

## Metrics

| Path                 | Metric           | HEAD^   | workspace   | Change   |
|----------------------|------------------|---------|-------------|----------|
| dvclive/metrics.json | epoch            | 10      | -           | -        |
| dvclive/metrics.json | step             | 890     | -           | -        |
| dvclive/metrics.json | test.epoch.acc   | 0.38351 | 0.37634     | -0.00717 |
| dvclive/metrics.json | test.epoch.f1    | 0.31615 | 0.29114     | -0.02501 |
| dvclive/metrics.json | test.step.acc    | 0.38351 | -           | -        |
| dvclive/metrics.json | test.step.f1     | 0.31615 | -           | -        |
| dvclive/metrics.json | train.epoch.acc  | 0.96381 | 1.0         | 0.03619  |
| dvclive/metrics.json | train.epoch.f1   | 0.9572  | 1.0         | 0.0428   |
| dvclive/metrics.json | train.epoch.loss | 0.10953 | -           | -        |
| dvclive/metrics.json | train.step.acc   | 1.0     | -           | -        |
| dvclive/metrics.json | train.step.f1    | 1.0     | -           | -        |
| dvclive/metrics.json | train.step.loss  | 0.01936 | -           | -        |
| dvclive/metrics.json | val.epoch.acc    | 0.91378 | -           | -        |
| dvclive/metrics.json | val.epoch.f1     | 0.906   | -           | -        |
| dvclive/metrics.json | val.epoch.loss   | 0.33504 | -           | -        |
| dvclive/metrics.json | val.step.acc     | 0.85714 | -           | -        |
| dvclive/metrics.json | val.step.f1      | 0.66667 | -           | -        |
| dvclive/metrics.json | val.step.loss    | 0.39189 | -           | -        |

# Plots

TODO
